src/gretel_trainer/benchmark/comparison.py

```diff
@@ -42,304 +26,52 @@
-            raise BenchmarkException("Gretel Job does not exist")
-
-        return job
-
-    def cleanup(self) -> None:
-        self._project.delete()
-
-        if self.config.trainer:
-            for project in search_projects(query=self.config.project_display_name):
-                project.delete()
-
-    def _result_dict(self, run_key: RunKey) -> dict[str, Any]:
-        executor = self.executors[run_key]
-        model_name, dataset_name = run_key
-
-        train_time = executor.strategy.get_train_time()
-        generate_time = executor.strategy.get_generate_time()
-
-        total_time = train_time
-        if train_time is not None and generate_time is not None:
-            total_time = train_time + generate_time
-
-        return {
-            "Input data": dataset_name,
-            "Model": model_name,
-            "DataType": executor.strategy.dataset.datatype,
-            "Rows": executor.strategy.dataset.row_count,
-            "Columns": executor.strategy.dataset.column_count,
-            "Status": executor.status.value,
-            **{
-                score_name: executor.get_report_score(score_key)
-                for score_name, score_key in self._report_scores.items()
-            },
-            "Train time (sec)": train_time,
-            "Generate time (sec)": generate_time,
-            "Total time (sec)": total_time,
-        }
-
-    def _setup_gretel_run(
-        self,
-        dataset: Dataset,
-        model: GretelModel,
-        artifact_key: Optional[str],
-        trainer_project_index: int,
-    ) -> None:
-        run_key = _make_run_key(model, dataset)
-        run_identifier = run_key.identifier
-        strategy = self._set_gretel_strategy(
-            benchmark_model=model,
-            dataset=dataset,
-            run_identifier=run_identifier,
-            trainer_project_index=trainer_project_index,
-            artifact_key=artifact_key,
-        )
-        executor = Executor(
-            strategy=strategy,
-            run_identifier=run_identifier,
-            evaluate_project=self._project,
-            config=self.config,
-        )
-        self._gretel_executors[run_key] = executor
-
-    def _setup_custom_run(
-        self,
-        dataset: Dataset,
-        model: CustomModel,
-        artifact_key: Optional[str] = None,
-    ) -> None:
-        run_key = _make_run_key(model, dataset)
-        run_identifier = run_key.identifier
-        strategy = CustomStrategy(
-            benchmark_model=model,
-            dataset=dataset,
-            run_identifier=run_identifier,
-            config=self.config,
-            artifact_key=artifact_key,
-        )
-        executor = Executor(
-            strategy=strategy,
-            run_identifier=run_identifier,
-            evaluate_project=self._project,
-            config=self.config,
+def launch(
+    *,
+    jobs: list[tuple[DatasetTypes, ModelTypes]],
+    config: Optional[BenchmarkConfig] = None,
+) -> Session:
```

Comment:
I put `launch` in here for now, as `compare` and `launch` are using the same helpers.

It doesn't fit 100%, so I was thinking of extracting that to somewhere else, since `gretel_trainer.benchmark.comparison.compare` is documented/main interface, I think renaming that module wouldn't be a good idea.

Reply:
I think we've only documented importing `compare` directly from `gretel_trainer.benchmark`. In v2 that import is basically just an alias ([see here](https://github.com/gretelai/trainer/blob/b2/src/gretel_trainer/benchmark/__init__.py#L3)). In v1, the __init__ file [defines the public compare function](https://github.com/gretelai/trainer/blob/main/src/gretel_trainer/benchmark/__init__.py#L61-L67) and translates arguments + defaults to a config that gets passed to an "internal" function signature (in hindsight I'm not a fan of this and am glad it has changed).

So I think it's fair game to rename this module. First idea that comes to my head is `entrypoints.py`, maybe?

Reply:
Awesome, that's great you had these aliases in the main module, that makes it easier to move things around.
`entrypoints.py` sounds great!